### ISSUE TITLE
Can't get AutoJoinServer to work.

### ISSUE THREAD

**Issue by DunklerPhoenix** [samcm/synclounge#125]
Hello.
I'm trying to setup my own synclounge server, but the autojoin feature will not work.

ENV: Docker 
Image: starbix/synclounge:latest
Browser: Chrome Beta and Firefox Stable

```
DOMAIN | plex.kartoffel.cloud/lounge
webroot | /lounge
autoJoin | true
serverroot | /loungemaster
autoJoinServer | https://plex.kartoffel.cloud/loungemaster
autoJoinRoom |  
autoJoinPassword |
```

I need to set the domain with the webroot, otherwise the invitelinks will not work and the webroot needs the / before, otherwise I'll get redirected to the mainpage.
The root of the domain is used by another service.

These are my nginx configurations:
```
     location /loungemaster {
         proxy_pass http://synclounge:8089/loungemaster;
         proxy_http_version 1.1;
         proxy_set_header Upgrade $http_upgrade;
         proxy_set_header Connection "upgrade";
     }         
     location /lounge {
         proxy_pass http://synclounge:8088/lounge;
         proxy_http_version 1.1;
         proxy_set_header Upgrade $http_upgrade;
         proxy_set_header Connection "upgrade";
     }        
```

The page works fine except for the autojoin.

https://plex.kartoffel.cloud/lounge
https://plex.kartoffel.cloud/loungemaster

What can I do?

**Comment by DunklerPhoenix** [comment]
Little update.
I hardcorded the values to the settings.json and SettingsHelper.js files.
Now the autojoin works, IF the site is reloaded once. 
At the first load after the login you get the "choose your server"-screen, but if you reload the page once  you'll get a green bar on the bottom with "Connected to https://plex.kartoffel.cloud/loungemaster"

**Comment by Crypto90** [comment]
Current behaviour: If you login, the autoJoinServer doesn't get selected. If I reload the website once directly after logging in, the autoJoinServer does get selected.
Video:
https://recordit.co/CIaHPT527B

**Comment by sigmaprojects** [comment]
I've been having this issue as well, it's pretty much a show-stopper for me

**Comment by samuelkadolph** [comment]
I started trying to dig into this issue because I want to use this feature. It looks like it works on master but only in production mode. I'm using it with the docker image and even with the latest version it doesn't work there. I think I did find the issue. The websocket request losses the path you provide. I found it tries to do hit `/socket.io/?EIO=3&transport=polling&t=XXX` instead of `/slserver/socket.io/?EIO=3&transport=polling&t=XXX`. So as a workaround I used my nginx host to also proxy requests for `socket.io`. So my host looks like this.

```
 location /slserver {
    proxy_pass http://172.20.5.7:8089/slserver;
  }

  location /socket.io {
    proxy_pass http://172.20.5.7:8089/slserver/socket.io;
  }

  location / {
    proxy_pass http://172.20.5.7:8088;
  }
```

This makes autojoin work for me. And oddly enough it just says `Connected to` instead of showing the server URL. So there's some issue with [this](https://github.com/samcm/synclounge/blob/master/src/store/modules/synclounge.js#L106-L118) code losing the value passed in.
